LabExT/Experiments/StandardExperiment.py

```diff
@@ -213,49 +236,81 @@ def run(self):
                 self._experiment_manager.main_window.model.var_mm_pause.set(True)
-                msg = 'Error occurred during measurement: ' + repr(exc)
-                messagebox.showinfo('Measurement Error', msg)
+                msg = "Error occurred during measurement: " + repr(exc)
+                messagebox.showinfo("Measurement Error", msg)
                 self.logger.exception(msg)
                 save_file_ending = "_error.json"
             except SystemExit:
                 # log error to file
                 etype, evalue, _ = sys.exc_info()
-                data['error'] = OrderedDict()
-                data['error']['type'] = "Abort"
-                data['error']['desc'] = "Measurement aborted by user."
-                data['error']['traceback'] = traceback.format_exc()
+                data["error"] = OrderedDict()
+                data["error"]["type"] = "Abort"
+                data["error"]["desc"] = "Measurement aborted by user."
+                data["error"]["traceback"] = traceback.format_exc()
                 save_file_ending = "_abort.json"
 
             finally:
-
                 # clear live plots after experiment finished
                 while len(self.live_plot_collection) > 0:
                     self.live_plot_collection.remove(self.live_plot_collection[0])
 
                 # save instrument parameters again
-                data['instruments'] = measurement._get_data_from_all_instruments()
+                data["instruments"] = measurement._get_data_from_all_instruments()
 
                 # get measurement end timestamp
-                ts = str('{date:%Y-%m-%d_%H%M%S}'.format(date=datetime.datetime.now()))
-                data['timestamp end'] = ts
-                data['timestamp'] = ts
-                data['finished'] = True
+                ts = str("{date:%Y-%m-%d_%H%M%S}".format(date=datetime.datetime.now()))
+                data["timestamp end"] = ts
+                data["timestamp"] = ts
+                data["finished"] = True
 
                 # save current measurement's data on disk
                 data.save()
                 data.auto_save = False
                 final_path = save_file_path + save_file_ending
                 rename(data.file_path, final_path)
 
-                self.logger.info('Saved data of current measurement: %s to %s',
-                                 measurement.get_name_with_id(),
-                                 final_path)
+                self.logger.info(
+                    "Saved data of current measurement: %s to %s", measurement.get_name_with_id(), final_path
+                )
+
+                if current_todo.part_of_sweep:
+                    sweep_params = current_todo.sweep_parameters
+                    # update sweep information
+                    meas_mask = sweep_params["id"] == measurement.get_name_with_id()
+                    meas_index = sweep_params[meas_mask].index.to_list()[0]
```

Comment:
Is my understanding correct that you use `measurement.get_name_with_id()` to get a unique identifier for a measurement within a sweep?

This is dangerous, as the return value thereof only contains the last 5 digits of the uuid. This might lead to collisions (I've been there!). I suggest you use the full UUID to actually uniquely identify a measurement mask/index within a sweep.